### PR TITLE
add a central pressure tap to IBFE instrument panel

### PR DESCRIPTION
**Pull request by boyceg**
### IBAMR Pull Request Checklist
- [x] Does the test suite pass?
- [x] Was clang-format run?
- [x] Were relevant issues cited? Please remember to add `Fixes #12345` to close
      the issue automatically if we are fixing the problem.
- [x] Is this a change others will want to know about? If so, then has a
      changelog entry been added?
- [x] If new data structures have been added to a class then have they been set
      up appropriately for restarts? If so, ensure that the restart version
      number is incremented.
- [x] Does this change include a bug fix or new functionality? If so, a new test
      or tests should be added. New tests should run quickly (less than a minute
      in release mode). If possible, an older test should gain a new option so
      that we do not need to compile more test executables.
- [x] Did you (if your account has permission to do so) set relevant labels on
      GitHub for the pull request?

### CODE REVIEW

**Comment by drwells** [comment]
This logic for looping across the patch levels should work, but could we just do the following instead?
1. set `my_intersecting_patch_level = -1`
2. loop across all locally owned patches and levels, update `my_intersecting_patch_level` if it contains the point we want
3. do a max reduction over `my_intersecting_patch_level`
4. if `my_intersecting_patch_level == max(my_intersecting_patch_level)` then use `LEInteractor` and broadcast that result

**Comment by boyceg** [comment]
Sure.

**Comment by drwells** [comment]
Do we want to get this done ASAP? If so I can put together a test.

**Comment by boyceg** [comment]
A test would be great!

**Comment by boyceg** [comment]
(We also should make tests for the case where the nodeset “misses” some points along the perimeter for Charles’ PR.)

**Comment by drwells** [comment]
```

        IBFE/instrument_panel_01_3d.input --------------------------- RUN FAILED
test failed with output:
    P=00000:Program abort called in file ``/__w/IBAMR/IBAMR/ibtk/src/lagrangian/LEInteractor.cpp'' at line 2281
    P=00000:ERROR MESSAGE: 
    P=00000:Failed assertion: Q_size / Q_depth == X_size / X_depth
```
looks like we have a few other things to fix too.

**Comment by boyceg** [comment]
@drwells: I have made the requested changes. I am also updating the way that quadrature point assignments are made for the meter meshes, to align with the way that it is done for the central point. I will make that a separate commit --- I am not sure it is clearly giving a reduction in complexity. 😁

**Comment by boyceg** [comment]
I guess that one advantage of this approach is that it could work with fully distributed Cartesian grid data structures (which would be available in SAMRAI 3/4).

**Comment by boyceg** [comment]
@drwells: Looks like all tests are passing. Let me know what you think about the second commit.

File: src/IB/IBFEInstrumentPanel.cpp
Comment (drwells) [inline src/IB/IBFEInstrumentPanel.cpp:356]: it would be nice to re-use FEDataManager here but I don't think its flexible enough - see also https://github.com/drwells/fiddle/pull/84

File: src/IB/IBFEInstrumentPanel.cpp
Comment (drwells) [inline src/IB/IBFEInstrumentPanel.cpp:432]: While this is correct I would still prefer to broadcast unique values instead of doing a reduction. Perhaps we can improve this later.

**Review by drwells — APPROVED** [review]
The extra changes to the quadrature point indexing should be OK since the tests pass - did this absorb @cpuelz's patch?

Lets add a test for the pressure before we merge.

File: src/IB/IBFEInstrumentPanel.cpp
Comment (boyceg) [reply src/IB/IBFEInstrumentPanel.cpp]: NP. I just made this change.

**Comment by boyceg** [comment]
> The extra changes to the quadrature point indexing should be OK since the tests pass - did this absorb @cpuelz's patch?
> 
> Lets add a test for the pressure before we merge.

No, this does not include @cpuelz's patch (#1508). I am going to take a look at that code separately.

**Comment by drwells** [comment]
Do you want to write the test or should I handle it?

**Comment by boyceg** [comment]
If you could write or update a test, that would be wonderful. I don't think I have enough time today both to write a test for this and also to review #1508.

**Comment by drwells** [comment]
I can wrap this up this afternoon (I need to go teach right now). The CI isn't happy about some signed vs. unsigned comparisons - I can fix those too.

**Comment by boyceg** [comment]
@drwells I will force push a fix for the comparison warning.

**Comment by boyceg** [comment]
> @drwells I will force push a fix for the comparison warning.

Comparison warning is fixed now. (I need to figure out why I'm not seeing these things in my editor...)

**Comment by drwells** [comment]
My best guess is that that particular CI image is using a newer and pickier version of GCC.

**Comment by boyceg** [comment]
> My best guess is that that particular CI image is using a newer and pickier version of GCC.

It turns out that I need to add `-Wextra` to get those warnings to show up in my UI.

**Comment by drwells** [comment]
Sometimes we don't calculate the meter value correctly in parallel - I'm investigating.

**Comment by drwells** [comment]
I fixed the broadcast bug and added three tests - this should be good to go once the CI check is complete.

**Pull request merged**
